src/components/plots/abstractPlot.ts

```diff
@@ -421,16 +427,59 @@ export module Plot {
       this._additionalPaint(maxTime);
     }
 
-    public getAllSelections(): D3.Selection {
-      var allSelections = d3.select();
-      allSelections[0] = [];
+    /**
+     * Retrieves all of the selections of this plot for the specified dataset(s)
+     *
+     * @param {string | string[]} datasetKeys The dataset(s) to retrieve the selections from.
+     * If not provided, all selections will be retrieved.
+     * @returns {D3.Selection} The retrieved selections.
+     */
+    public getAllSelections(datasetKeys?: string | string[]): D3.Selection {
+      var datasetKeyArray: string[] = [];
+      if (datasetKeys == null) {
+        datasetKeyArray = this._datasetKeysInOrder;
+      } else if (typeof(datasetKeys) === "string") {
+        datasetKeyArray = [<string> datasetKeys];
+      } else {
+        datasetKeyArray = <string[]> datasetKeys;
+      }
+
```

Comment:
As per my comment on #1659, we shouldn't be special-casing out the `Line` drawer here -- instead we should have `Plot.Line` override this method with a `Line`-specific implementation.